cuqi/implicitprior/_regularizedUniform.py

```diff
@@ -0,0 +1,61 @@
+from cuqi.implicitprior import RegularizedGaussian
+from cuqi.distribution import Distribution, Gaussian
+
+import numpy as np
+
+
+class RegularizedUniform(RegularizedGaussian):

```

Comment:
I wonder if using "uniform" in the name can be misleading. A conventional uniform distribution will live only on a bounded domain with some finite value, whereas this lives on the entire space.  Would there be any sense in the name "RegularizedFlatGaussian" or even "RegularizedFlat". Here, "flat" replaces "uniform" to convey it is, well, flat, and detaches from the usual bounded domain associated with "uniform".  "gaussian" would keep the connection to the underlying nature of the class.  The second option "RegularizedFlat" is less clear and a bit odd, with the main attractive aspect being its shortness. Thoughts?

Reply:
My two kroner on the matter:

As for "RegularizedFlatGaussian":
 The reason the current RegularizedUniform class inherits from RegularizedGaussian is to make RegularizedLinearRTO work without any modifications. However, this is far from optimal as computations are being done with an all zero matrix (the sqrtprec). It could be that the RegularizedLinearRTO and the current RegularizedUniform might be reworked in the future to not require this anymore, resulting in there no longer being any underlying Gaussian. It is with that argument in mind that I prefer not to refer to this implicit prior with the Gaussian suffix.

Regarding "RegularizedUniform" or "RegularizedFlat":
 I agree that a uniform distribution can only _properly_ exist on a bounded domain, however, within the Bayesian literature, there is distinction between _proper_ and _improper_ priors, based on whether the prior represents an actual probability distribution or not. A uniform prior on a unbounded domain (or more precisely on a domain of infinite measure) is a commonly used example of such an improper prior. In CUQIpy, as far as I am aware, you can create such improper priors already, for example, by creating a Gaussian "distribution" with a sqrtprec that is not full-rank. This is precisely the trick I currently use to make the RegularizedUniform class work. It is wat that argument in mind that I prefer to refer to this implicit prior with the Uniform suffix.

I hope this clarifies my choice for the naming decision a little bit, and perhaps spark a little bit of a discussion on CUQIpy conventions for proper and improper priors.

Reply:
How about RegularizedUnboundedUniform?

Reply:
I am fine with that, how about you @jakobsj ?

Reply:
Many thanks for the background. I agree RegularizedUnboundedUniform is a sensible name.